src/DateTimeGuess/Assigners/Assigner.cs

```diff
@@ -51,12 +51,15 @@ public Assigner(string name, string type, Format format)
         /// <returns>Return the <see cref="Token"/>.</returns>
         public virtual Token Assign(Token token)
         {
-            foreach (KeyValuePair<Regex, string> kvp in Map)
+            if (TestTokenType(token))
             {
-                if (TestTokenType(token) && kvp.Key.Match(token.Value).Success)
```

Comment:
Is there a reason for breaking this up?

Reply:
It's faster to put that check before the loop of the Map . Then, you don't have to do that check for each value in the Map.